### DISCUSSION TITLE
Can we configure `restart_on` for a process?

### DISCUSSION THREAD

**Discussion by shivaraj-bh** [F1bonacc1/process-compose#247] [Q&A] [answered]
`restart_on` can define a condition/s under which a process has to be restarted.

Use case:
If there are two dependent processes `A` and `B`, where B `depends_on` `A`. I would like `B` to also restart when `A` does. 

I don’t see any straightforward way to configure this for now, is this worth exploring as a new feature?

**Comment by F1bonacc1** [comment]
Hey @shivaraj-bh,

Can you elaborate a bit more on the reason for A's restart?

I asking because I am considering if something like liveness probe can help here.

**Reply by shivaraj-bh** [reply]
Yes, I should’ve added more details.

Let’s say `A` is a process that builds a package, while `B` is used to run it. `B` depends on `A` being successfully completed and `B` also has `readiness_probe` configured. In a local development scenario, I would want `B` to automatically restart (based on a configured policy) when I restart `A`.

Note: I can’t just do build and run in the same process because then the `readiness_probe` will close the process after `failure_threshold`, leading to the process terminating if the build isn’t complete yet.

**Reply by F1bonacc1** [reply, marked as answer]
That clears things up.
Another question - how do you restart A? Do you do it manually? Externally or through Process Compose?
Is it possible to add another process to PC, let's call it "restarter" that will call `process-compose process restart A && process-compose process restart B`?

**Reply by shivaraj-bh** [reply]
> Is it possible to add another process to PC, let's call it "restarter" that will call process-compose process restart A && process-compose process restart B

Right, that should do the job. Thank you!
